src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -645,10 +648,20 @@ def get_result_set(
                 final_queries = []
                 last_place_holder = None
                 for q in main_queries:
+                    query = q.sql
+                    if q.temp_name_place_holder:
```

Comment:
instead of doing it here, might be better to do it at the execute_queries when calling plan_compiler to get the final query, so that this can be applied to all user facing queries